### PR TITLE
Fix being able to minimise to systray on systems without a systray

### PR DESCRIPTION
**Pull request by ForeverRainbow**
This is a fix for #926 on the C++ version. It disables the option to enable docking to the system tray and forces normal startup on desktop environments without system trays (e.g gnome). 

I have tested it on a windows 10 machine and the system tray works fine there (the option isnt disabled and it minimises)

### CODE REVIEW

File: future/src/ct/ct_main_win.cc [hunk 1]
```diff
@@ -124,10 +124,17 @@ CtMainWin::CtMainWin(bool             start_hidden,
     menu_set_items_special_chars();
     _uCtMenu->find_action("ct_vacuum")->signal_set_visible.emit(false);
 
-    if (start_hidden || (_pCtConfig->systrayOn && _pCtConfig->startOnSystray))
-        set_visible(false);
-    else
+    if (start_hidden || (_pCtConfig->systrayOn && _pCtConfig->startOnSystray)) {
```
Comment (txe) [inline future/src/ct/ct_main_win.cc:127]: start_hidden is used to hide windows during import from command line. It's not related to tray, so it has to work

**Comment by txe** [comment]
'menu_set_visible_exit_app' should be called with false when no systray

**Comment by ForeverRainbow** [comment]
@txe If there is no systray then it will be anyway? Since the user cannot enable the option?

**Comment by txe** [comment]
all menu items are created visible by default. By calling this function, we control visibility of App Exit (menu item); we should hide it if not systray

File: future/src/ct/ct_pref_dlg.cc [hunk 2]
```diff
@@ -1343,6 +1343,11 @@ Gtk::Widget* CtPrefDlg::build_tab_misc()
     Gtk::VBox* vbox_system_tray = Gtk::manage(new Gtk::VBox());
     Gtk::CheckButton* checkbutton_systray = Gtk::manage(new Gtk::CheckButton(_("Enable System Tray Docking")));
     Gtk::CheckButton* checkbutton_start_on_systray = Gtk::manage(new Gtk::CheckButton(_("Start Minimized in the System Tray")));
+    bool has_systray = _pCtMainWin->get_status_icon()->is_embedded();
+    checkbutton_systray->set_sensitive(has_systray);
+    if (!has_systray) {
+        checkbutton_systray->set_tooltip_text("Your system does not appear to support system trays");
```
Comment (giuspen) [inline future/src/ct/ct_pref_dlg.cc:1349]: this should be translated, so set_tooltip_text(_("Your system does not appear to support system trays"));

**Comment by txe** [comment]
Now I see why you asked, you skipped another  `menu_set_visible_exit_app`

**Comment by ForeverRainbow** [comment]
Where? Couldn't find it in `ct_main_win`

**Comment by txe** [comment]
in `void CtMainWin::config_apply()`, line before the last one

**Comment by ForeverRainbow** [comment]
Sorry, had to do a thing. OK fixed it.

**Comment by txe** [comment]
That's OK, thanks

**Pull request merged**